include/net/socket.h

```diff
@@ -370,6 +370,20 @@ int zsock_getsockopt(int sock, int level, int optname,
 int zsock_setsockopt(int sock, int level, int optname,
 		     const void *optval, socklen_t optlen);
 
+/**
+ * @brief Get socket name
+ *
+ * @details
+ * @rststar
+ * See `POSIX.1-2017 article
+ * <http://pubs.opengroup.org/onlinepubs/9699919799/functions/getsockname.html>`__
+ * for normative description.
+ * This function is also exposed as ``getsockname()``
+ * if :option:`CONFIG_NET_SOCKETS_POSIX_NAMES` is defined.
+ * @endrststar
+ */
+int zsock_getsockname(int sock, struct sockaddr *addr, socklen_t *addrlen);
```

Comment:
isn't the syscall version missing?

Reply:
@tbursztyka right! I added the `__syscall` attribute (although it just defines it as `static inline` as far as I can see?). Is that what you had in mind?

BTW. When should the definitions using `Z_SYSCALL_HANDLER` be added? Should I add one too? Are there any guidelines for that? How do I test it after adding it?

Reply:
You need to add a Z_SYSCALL_HANDLER in this case yes.
I guess the rule is: when it's an API meant to be accessible from user space, and when it access objects in kernel space, then you'll need to have this syscall handler.

Reply:
Ok, thanks! I just added the syscall handler (tested it by combining the userspace demo - shared_mem and the sockets demo)